browser/js/common/directives/navbar/navbar.js

```diff
@@ -1,47 +1,71 @@
-app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state) {
+app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state, $window) {
 
-    return {
-        restrict: 'E',
-        scope: {},
-        templateUrl: 'js/common/directives/navbar/navbar.html',
+	return {
+		restrict: 'E',
+		scope: {},
+		templateUrl: 'js/common/directives/navbar/navbar.html',
 
-        link: function (scope) {
+		link: function (scope) {
+			var temp = 0;
+			scope.goToSection = function(val, state) {
+				if (val !== 'null') {
+					Promise.resolve()
+					.then( () => {
+						if (!state) {
+							state = 'home';
+						}
+						if($state.current.name === state) {
+							return;
+						} else {
+						return $state.go(state);
+						}
+					})
+					.then( () => {
+						// temp = $window.pageYOffset;
+						// $window.pageYOffset -= 58;
+						smoothScroll(document.querySelector(val));
+					});
+					// .then( () => {
+						// setTimeout(function() {
+							// $window.pageYOffset = window.scrollY;
+						// } , 500);
+					// });
+				} else {
+					$state.go(state);
+				}
+			};
 
-            scope.goToSection = function(val) {
-                smoothScroll(document.querySelector(val));
-            };
+			scope.user = null;
```

Comment:
I'm not sure. I'll look into it. Maybe using something like hunks to split it up would work.